common/src/main/java/com/wynntils/wynn/model/questbook/QuestInfo.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.wynn.model.questbook;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.mc.utils.ItemUtils;
+import com.wynntils.utils.Pair;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.world.item.ItemStack;
+
+public class QuestInfo {
+    private static final Pattern QUEST_NAME_MATCHER = Pattern.compile("^§.§l([^֎À]*)[֎À]+ $");
+    private static final Pattern STATUS_MATCHER = Pattern.compile("^§.(.*)(?:\\.\\.\\.|!)$");
+    private static final Pattern LENGTH_MATCHER = Pattern.compile("^§a-§r§7 Length: §r§f(.*)$");
+    private static final Pattern LEVEL_MATCHER = Pattern.compile("^§..§r§7 Combat Lv. Min: §r§f(\\d+)$");
+    private static final Pattern REQ_MATCHER = Pattern.compile("^§..§r§7 (.*) Lv. Min: §r§f(\\d+)$");
+
+    private final String name;
+    private final QuestStatus status;
+    private final QuestLength length;
+    private final int level;
+    private final String nextTask;
+    /** Additional requirements as pairs of <"profession name", minLevel> */
+    private final List<Pair<String, Integer>> additionalRequirements;
+
+    public QuestInfo(
+            String name,
+            QuestStatus status,
+            QuestLength length,
+            int level,
+            String nextTask,
+            List<Pair<String, Integer>> additionalRequirements) {
+        this.name = name;
+        this.status = status;
+        this.length = length;
+        this.level = level;
+        this.nextTask = nextTask;
+        this.additionalRequirements = additionalRequirements;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public QuestStatus getStatus() {
+        return status;
+    }
+
+    public QuestLength getLength() {
+        return length;
+    }
+
+    public int getLevel() {
+        return level;
+    }
+
+    public String getNextTask() {
+        return nextTask;
+    }
+
+    public List<Pair<String, Integer>> getAdditionalRequirements() {
+        return additionalRequirements;
+    }
+
+    @Override
+    public String toString() {
+        return "QuestInfo[" + "name=\""
+                + name + "\", " + "status="
+                + status + ", " + "length="
+                + length + ", " + "minLevel="
+                + level + ", " + "nextTask=\""
+                + nextTask + "\", " + "additionalRequirements="
+                + additionalRequirements + "]";
+    }
+
+    public static QuestInfo parseItem(ItemStack item) {
```

Comment:
The defensive programming here :\)

Reply:
I've tried my best not to lose any more mythics. :)